ci/scripts/cpp_checks.sh

```diff
@@ -80,9 +80,8 @@ if [[ -n "${MRC_MODIFIED_FILES}" ]]; then
 
    # Include What You Use
    if [[ "${SKIP_IWYU}" == "" ]]; then
-      # Remove .h, .hpp, and .cu files from the modified list
       shopt -s extglob
-      IWYU_MODIFIED_FILES=( "${MRC_MODIFIED_FILES[@]/*.@(h|hpp|cu)/}" )
+      IWYU_MODIFIED_FILES=( "${MRC_MODIFIED_FILES[@]}" )
```

Comment:
True, but right below this part, we combine the list of changed files with only the compilation units found in `compile_commands.json`. Since we only include our local `.cpp` files in this file, there is no chance for it to get headers or `.cu` files.

Turns out, that what this was doing wasnt removing items from the array, but instead just blanking them out. If you happened to have the first file changed be a `.h` file, then IWYU was skipped entirely